pkg/types/position.go

```diff
@@ -453,3 +457,163 @@ func (p *Position) AddTrade(td Trade) (profit fixedpoint.Value, netProfit fixedp
 
 	return fixedpoint.Zero, fixedpoint.Zero, false
 }
+
+type PositionStack struct {
+	*Position
+	Stack []*Position
+}
+
+func (stack *PositionStack) Push(pos *Position) *PositionStack {
+	stack.Position = pos
+	stack.Stack = append(stack.Stack, pos)
+	return stack
+
+}
+
+func (stack *PositionStack) Pop() *PositionStack {
+	if len(stack.Stack) < 1 {
+		return nil
+	}
+	stack.Position = stack.Stack[len(stack.Stack)-1]
+	stack.Stack = stack.Stack[:len(stack.Stack)-1]
+	return stack
+}
+
+func NewPositionStackFromMarket(market Market) *PositionStack {
+	pos := &Position{
+		Symbol:        market.Symbol,
+		BaseCurrency:  market.BaseCurrency,
+		QuoteCurrency: market.QuoteCurrency,
+		Market:        market,
+		TotalFee:      make(map[string]fixedpoint.Value),
+	}
+	return &PositionStack{
+		Position: pos,
+		Stack:    []*Position{pos},
+	}
+}
+
+func (p *PositionStack) AddTrade(td Trade) (profit fixedpoint.Value, netProfit fixedpoint.Value, madeProfit bool) {
```

Comment:
but I guess you don't need to re-implement this method here?

Reply:
that's true done